repository/git.go

```diff
@@ -272,7 +274,7 @@ func (r *Repository) saveState(ctx context.Context, env *environment.Environment
 		return err
 	}
 
-	return r.lockManager.WithLock(ctx, LockTypeGitNotes, func() error {
+	return r.lockManager.WithLock(ctx, LockTypeNotes, func() error {
```

Comment:
Quick question: Right above, we do a `defer f.Close()`

Would it make sense to clean up the temp file explicitly after the `git notes add -F` operation?

I was just thinking, for longer-running sessions, this might help avoid leaving unnecessary temp files in the OS tmp directory ?